### ISSUE TITLE
SpicesUpdate@claudiux not working

### ISSUE THREAD

**Issue by Padre2** [linuxmint/cinnamon-spices-applets#5682]
### Applet version/Build date

7.0.0

### Cinnamon version

6.0.4

### Distribution

Arch 6.8.4

### Graphics hardware *and* driver used

Radeon HD5450

### Applet name and maintainer

SpicesUpdate@claudiux @claudiux

### What happened?

Downloaded and installed manually.
Applet displays in list showing version 7.0.0 details as expected, but configuration button not active,
Restarted Cinnamon without effect.
No icon in panel, nothing in Looking Glass.

**Comment by claudiux** [comment]
Have you added this applet to a panel? The configuration button is not active if you haven't done so.

**Comment by Padre2** [comment]
Apologies, it's been years since I added new applets and rarely manually. I hadn't clicked the + to the left of the applet and,  after clicking it, the configuration button became active and there is an icon in the left panel, but no icon in the right panel.

Clicking the icon the Spices Update window opens the General Settings, which can be changed.

No applets, desklets. extensions or themes are listed. Clicking the Refresh  and Open Cinnamon Settings buttons have no results.
Closing the Spices Update Window removes the icon in the left panel.

Am I still missing something?

**Comment by claudiux** [comment]
To move the applet, right-click on the panel, select "Panel edit mode", drag the applet into its new location, right-click on the panel, deselect "Panel edit mode".

**Comment by Padre2** [comment]
The icon in the left panel is the slightly larger always coloured icon that appears only after clicking the configuration button. It cannot be dragged to the right panel. It disappears after the configuration window is closed.

**Comment by claudiux** [comment]
This is the icon for the configuration window. It is normal for it to disappear once this window is closed.

**Comment by claudiux** [comment]
I'm currently writing the next version. I hope it will solve your problem.

**Comment by Padre2** [comment]
Thank you.

**Comment by claudiux** [comment]
Oh, you're on Arch.
Please execute in a terminal:
`~/.local/share/cinnamon/applets/SpicesUpdate@claudiux/scripts/install_symbola_on_Arch.sh`
Then restart Cinnamon.

**Comment by Padre2** [comment]
After running the above script and restarting Cinnamon, there was no change: no icon was added, but a warning message appeared above the right panel referring to errors log file and Looking Glass.

.xsession-errors

[SpicesUpdate@claudiux] 10:48:26: now=1712893707 ; this.general_next_check_date=1712893767
Gjs-Message: 10:48:26.966: JS LOG: [LookingGlass/error] 
[SpicesUpdate@claudiux]: GSettings schema com.linuxmint.updates not found
[SpicesUpdate@claudiux]: Failed to evaluate 'main' function on applet: SpicesUpdate@claudiux/225
Gjs-Message: 10:48:26.967: JS LOG: [LookingGlass/trace] 
<----------------
_init@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:753:23
get_blacklisted_packages@/home/victor/.local/share/cinnamon/applets/SpicesUpdate@claudiux/6.0/applet.js:1913:35
populateSettingsUnprotectedSpices@/home/victor/.local/share/cinnamon/applets/SpicesUpdate@claudiux/6.0/applet.js:1124:34
SpicesUpdate@/home/victor/.local/share/cinnamon/applets/SpicesUpdate@claudiux/6.0/applet.js:261:18
main@/home/victor/.local/share/cinnamon/applets/SpicesUpdate@claudiux/6.0/applet.js:2710:12
createApplet@/usr/share/cinnamon/js/ui/appletManager.js:597:25
addAppletToPanels@/usr/share/cinnamon/js/ui/appletManager.js:372:34
finishExtensionLoad@/usr/share/cinnamon/js/ui/appletManager.js:98:14
_init/<@/usr/share/cinnamon/js/ui/extension.js:286:32
---------------->
Gjs-Message: 10:48:26.968: JS LOG: [LookingGlass/error] 
[SpicesUpdate@claudiux]: Applet SpicesUpdate@claudiux: Could not create applet object.
[SpicesUpdate@claudiux]: Error importing applet.js from SpicesUpdate@claudiux
Gjs-Message: 10:48:26.968: JS LOG: [LookingGlass/trace] 
<----------------
_init/<@/usr/share/cinnamon/js/ui/extension.js:287:23
---------------->

(cinnamon:823): Gjs-CRITICAL **: 10:48:27.012: JS ERROR: Error: GSettings schema com.linuxmint.updates not found
_init@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:753:23
get_blacklisted_packages@/home/victor/.local/share/cinnamon/applets/SpicesUpdate@claudiux/6.0/applet.js:1913:35
populateSettingsUnprotectedSpices@/home/victor/.local/share/cinnamon/applets/SpicesUpdate@claudiux/6.0/applet.js:1124:34
populateSettingsUnprotectedDesklets@/home/victor/.local/share/cinnamon/applets/SpicesUpdate@claudiux/6.0/applet.js:1199:14
bind/<@resource:///org/gnome/gjs/modules/script/lang.js:71:25
_checkSettings@/usr/share/cinnamon/js/ui/settings.js:611:45

Melange

[SpicesUpdate@claudiux] 10:48:26: now=1712893707 ; this.general_next_check_date=1712893767
error t=2024-04-12T10:48:26Z 
[SpicesUpdate@claudiux]: GSettings schema com.linuxmint.updates not found
[SpicesUpdate@claudiux]: Failed to evaluate 'main' function on applet: SpicesUpdate@claudiux/225
error t=2024-04-12T10:48:26Z 
[SpicesUpdate@claudiux]: Applet SpicesUpdate@claudiux: Could not create applet object.
[SpicesUpdate@claudiux]: Error importing applet.js from SpicesUpdate@claudiux

Btw, you are now referring to version 7.1 but no update is showing in the downloads.

**Comment by Padre2** [comment]
After running the above script and restarting Cinnamon, there was no change: no icon was added, but a warning message appeared above the right panel referring to errors log file and Looking Glass.

I copied the messages and posted, but after rereading your last posting, I noticed that it referred to version 7.1. I uninstalled version 7.0 and reinstalled the new version 7.1.1. Still no icon and not working.

I deleted my previous posting and copied the error messages again as below:

.xsession-errors

Gjs-Message: 12:13:06.788: JS LOG: [LookingGlass/error] 
[SpicesUpdate@claudiux]: GSettings schema com.linuxmint.updates not found
[SpicesUpdate@claudiux]: Failed to evaluate 'main' function on applet: SpicesUpdate@claudiux/226
Gjs-Message: 12:13:06.788: JS LOG: [LookingGlass/trace] 
<----------------
_init@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:753:23
get_blacklisted_packages@/home/victor/.local/share/cinnamon/applets/SpicesUpdate@claudiux/6.0/applet.js:1911:35
populateSettingsUnprotectedSpices@/home/victor/.local/share/cinnamon/applets/SpicesUpdate@claudiux/6.0/applet.js:1113:32
SpicesUpdate@/home/victor/.local/share/cinnamon/applets/SpicesUpdate@claudiux/6.0/applet.js:261:18
main@/home/victor/.local/share/cinnamon/applets/SpicesUpdate@claudiux/6.0/applet.js:2714:12
createApplet@/usr/share/cinnamon/js/ui/appletManager.js:597:25
addAppletToPanels@/usr/share/cinnamon/js/ui/appletManager.js:372:34
finishExtensionLoad@/usr/share/cinnamon/js/ui/appletManager.js:98:14
_init/<@/usr/share/cinnamon/js/ui/extension.js:286:32
---------------->
Gjs-Message: 12:13:06.790: JS LOG: [LookingGlass/error] 
[SpicesUpdate@claudiux]: Applet SpicesUpdate@claudiux: Could not create applet object.
[SpicesUpdate@claudiux]: Error importing applet.js from SpicesUpdate@claudiux
Gjs-Message: 12:13:06.790: JS LOG: [LookingGlass/trace] 
<----------------
_init/<@/usr/share/cinnamon/js/ui/extension.js:287:23
---------------->
Gjs-Message: 12:13:06.791: JS LOG: [LookingGlass/info] AppletManager started in 11398 ms
Gjs-Message: 12:13:06.803: JS LOG: [LookingGlass/info] Cinnamon took 14729 ms to start

(cinnamon:824): Gjs-CRITICAL **: 12:13:06.810: JS ERROR: Error: GSettings schema com.linuxmint.updates not found
_init@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:753:23
get_blacklisted_packages@/home/victor/.local/share/cinnamon/applets/SpicesUpdate@claudiux/6.0/applet.js:1911:35
populateSettingsUnprotectedSpices@/home/victor/.local/share/cinnamon/applets/SpicesUpdate@claudiux/6.0/applet.js:1113:32
populateSettingsUnprotectedDesklets@/home/victor/.local/share/cinnamon/applets/SpicesUpdate@claudiux/6.0/applet.js:1197:14
bind/<@resource:///org/gnome/gjs/modules/script/lang.js:71:25
_checkSettings@/usr/share/cinnamon/js/ui/settings.js:611:45

Melange

error t=2024-04-12T12:13:06Z 
[SpicesUpdate@claudiux]: GSettings schema com.linuxmint.updates not found
[SpicesUpdate@claudiux]: Failed to evaluate 'main' function on applet: SpicesUpdate@claudiux/226
error t=2024-04-12T12:13:06Z 
[SpicesUpdate@claudiux]: Applet SpicesUpdate@claudiux: Could not create applet object.
[SpicesUpdate@claudiux]: Error importing applet.js from SpicesUpdate@claudiux

**Comment by claudiux** [comment]
> GSettings schema com.linuxmint.updates not found

This is the source of the problem. I'm searching on Arch...

**Comment by claudiux** [comment]
Found! This error occurs because Arch (and many other distribs) doesn't use mintupdate. Problem solved, I make a new PR soon.

**Comment by claudiux** [comment]
@Padre2 Please update this applet to version 7.1.2.

**Comment by Padre2** [comment]
@claudiux During the attempts to get Spices Update working, the xsession-errors file has been flooded with issues and Cinnamon has been behaving abnormally.  Somewhere above you had told me to install Soup 3 and while I was trying to find a repository, Cinnamon crashed the computer. Although I was able to restart, the Cinnamon issue repeated and the terminal also stopped working.  After several attempts to restart, the problems increased to where it dropped to rootfs. I compiled this Arch installation more than 3-1/2 years ago as my first real entrée into Linux and it has performed without any issues that I couldn't fix quickly. This has never happened before, so it will take me time to sort it out, if at all, or perhaps reinstall, assuming my back-up is usable. I will let you know when I can try the applet again. Thank you for all your hard work.

**Comment by claudiux** [comment]
@Padre2 So sorry for the inconvenience! Remove this applet completely. If you can show here the entire contents of your `~/.xsession-errors` file, I would like to understand...

**Comment by Padre2** [comment]
Arch would not boot and, after trying to recover, I got the message that _fsck_ should be run manually without _-a_ . Since I have never needed to do that for almost 4 years, and it was too late last night to get my old brain in gear, I decided to sleep on it. This morning the cursor ended up again at  _rootfs_.  I guessed the settings and tried _fsck_ without any flags:  _fsck /dev/sda1_ and it ran finding many errors, which required my input.  It then went into pages of numbers that ran for tens of minutes, then after stopping required my input (**_y_**) to fix. I just held my finger on **_y_** until completed and the process repeated. During this time, I regretted not having done more research to see whether I could have safely used flags to simplify the process. On completion I rebooted and, to my surprise, Arch booted as if nothing had ever happened - incroyable. I deleted the previous SpicesUpdate and installed version 3 and everything worked normally including several Spices Updates. The robustness of my Arch compilation never ceases to amaze me.

**Comment by Padre2** [comment]
@claudiux  I do not blame the applet, but when I restarted Cinnamon after updating the applet to version 7.1, it immediately crashed the system, requiring a power-off.  The xsessions-error file increased from about 9 kB to 49 KB and restarting Cinnamon crashed the system again after which it wouldn't boot. Power-offs are very undesirable and presumably corrupted the file system, but as Shakespeare wrote _all's well that ends well_. Now I will try to find out what happened to the Google Calendar desklet

**Comment by claudiux** [comment]
How many RAM do you have on your computer? Is your swap frequently used?

**Comment by Padre2** [comment]
I have 8 GB RAM. The swap file is 4.3 GB.
The compilation and settings including a LAMP server have existed for 4 years.
I presume the swap file is used more when I have a lot of tabs open on browsers and I open all my development sites on LAMP.
With 2 browsers and several tabs open on each, I ran  `free -h`
`               total       used        free       shared    buff/cache   available`
`Mem:           7.8Gi       4.0Gi       1.6Gi       108Mi       2.5Gi       3.8Gi`
`Swap:          4.0Gi       524Ki       4.0Gi`

**Comment by claudiux** [comment]
Mmmh... Your swap is used when you have enough mem available.
You should reset your swap and reduce the swappiness of your computer. Also increase the `fs.inotify.max_user_watches` value.

To do all that:

1.  Reset your swap: `sudo swapoff -a`
2. Reactivate your swap: `sudo swapon -a`
3. To know your current swappiness: `cat /proc/sys/vm/swappiness`
4. Reduce the swappiness and increase the fs.inotify.max_user_watches value: `sudo xed /etc/sysctl.d/99-sysctl.conf`
At the end of this file, add these two lines:
```
vm.swappiness=30
fs.inotify.max_user_watches=1000000
```
(You can choose your own values. 30 means your computer begins to use the swap when the amount of free mem is less than 30%.)
5. Save and close this file, then run: `sudo sysctl -p /etc/sysctl.d/99-sysctl.conf` to take into account these new parameters.

**Comment by Padre2** [comment]
I have 8 GB RAM. The swap file is 4.3 GB.
The compilation and settings including a LAMP server have existed for 4 years.
I presume the swap file is used more when I have a lot of tabs open on browsers and I open all my development sites on LAMP.
With 2 browsers and several tabs open on each, I ran  `free -h`
`               total       used        free       shared    buff/cache   available`
`Mem:           7.8Gi       4.0Gi       1.6Gi       108Mi       2.5Gi       3.8Gi`
`Swap:          4.0Gi       524Ki       4.0Gi`

**Comment by Padre2** [comment]
Thank you for your advice about the swap file.
Since I had not encountered memory problems, I have never played with the swap file settings. 
I ran `cat /proc/sys/vm/swappiness` which showed 60%.
I think `xed` is Mint and ran `sudo gedit /etc/sysctl.d/99-sysctl.conf vm.swappiness=30 fs.inotify.max_user_watches=1000000`.
A `99-sysctl.conf` file opened with the tabs 99-sysctl.conf, vm.swappiness =30 fs.inotify.max_user_watches=1000000 and I saved and closed it.

![Screenshot from 2024-04-16 20-10-22](https://github.com/linuxmint/cinnamon-spices-applets/assets/38849451/90cd8937-26ae-4d84-abf8-f8909863068f)

I then ran`sudo sysctl -p /etc/sysctl.d/99-sysctl.conf` and checked the swappiness, but it was still 60%.
Opening the 99-sysctl.conf file, it was empty.
I repeated the above with the same result, so I presume an error when saving.

**Comment by claudiux** [comment]
Don't use gedit but nano (CTRL+X to save and exit).

**Comment by Padre2** [comment]
Oh yes, was not thinking!  I ran with nano, but also it didn't save the lines. Checking `nano`, it appears it may need using Ctrl+O to save then Ctrl+X. It didn't work either, because it seemed to want to write 3 lines with the file name as the first. Gave up and copied the 2 lines into the file and saved. Ran `sudo sysctl -p /etc/sysctl.d/99-sysctl.conf` and checked the swappiness, now showing 30%. 
Should I assume that with my 8 GB of RAM,  30% is a better setting than the 60% that I was using?

**Comment by claudiux** [comment]
Try the Radio3.0@claudiux applet: https://cinnamon-spices.linuxmint.com/applets/view/360
It is in active development. Use it with sound150@claudiux: https://cinnamon-spices.linuxmint.com/applets/view/306. This is the perfect combo.

**Comment by Padre2** [comment]
Looks like a very well designed applet. I have just installed it and it should be a better replacement. I didn't get any sound yet, does it need to be paired with your sound applet, or should it work without it? I will continue the installation tomorrow.

**Comment by claudiux** [comment]
You should search about new stations, or restore the list `Radio3.0_EXAMPLES.json` from the Radios tab of settings.

**Comment by claudiux** [comment]
Have you tried SpicesUpdate version 7.3.3?

**Comment by Padre2** [comment]
SpicesUpdate updated yesterday and everything seems good.
Radio3.00 is excellent. The Search is helpful and I like the facility to check and update the data in the list, as many of my BBC URLs were not not working.

**Comment by claudiux** [comment]
@Padre2 
I'm glad you like all these applets. Please increase their score on their pages. Thank you for your Italian translations!
